git/git.go

```diff
@@ -43,6 +43,18 @@ type Ref struct {
 	Sha  string
 }
 
+// MatchesNameAndType returns whether or not the receiving *Ref "r" has the same
+// name and type as the "other" *Ref. Nil *Refs are invalid values and thus do
+// not possess equality.
+func (r *Ref) MatchesNameAndType(other *Ref) bool {
```

Comment:
Is there any (valid) situation where Name & Type are equal but Sha is not? Since we never modify refs I don't think so. In which case the struct equality operator is enough?